libmscore/score.cpp

```diff
@@ -2692,6 +2693,15 @@ void Score::cmdConcertPitchChanged(bool flag, bool /*useDoubleSharpsFlats*/)
       {
       undoChangeStyleVal(Sid::concertPitch, flag);       // change style flag
 
+      for (Segment* segment = firstSegment(SegmentType::Ambitus); segment; segment = segment->next1(SegmentType::Ambitus)) {
```

Comment:
I'd rather leave this loop early for all non-transposing instruments, but attempts to do so, like using code similar to the one in the below loop, leads to cashes when calculating the `Interval`.
Probably should leave early in case of a percussion staff too (again in the fashion the loop below does.

Reply:
Can we just add two Note objects into the innards of an Ambitus object, recreated upon reading from MSCX or creating the ambitus (i.e. transient, not stored by themselves), and set their properties to the respective highest and lowest note’s on update, and then use the usual transposition functions (`epitch()`, `ppitch()`, `tpc1()`, `tpc2()`, etc.) internally in the Ambitus’ functions?

Reply:
Code sample?

Reply:
I won't mind you to PR on top of mine ;-)

Reply:
Wish I had any… but that’s going to be something major, and I’m behind on billable hours as is. Sorry.